cg/cli/workflow/mip_dna/base.py

```diff
@@ -38,43 +37,42 @@
 @click.pass_context
 def mip_dna(context: click.Context, case_id: str, email: str, priority: str, start_with: str):
     """Rare disease DNA workflow"""
-    context.obj["db"] = Store(context.obj["database"])
-    hk_api = hk.HousekeeperAPI(context.obj)
-    scout_api = scoutapi.ScoutAPI(context.obj)
-    lims_api = lims.LimsAPI(context.obj)
-    context.obj["tb"] = tb.TrailblazerAPI(context.obj)
-    deliver = DeliverAPI(
-        context.obj, hk_api=hk_api, lims_api=lims_api, case_tags=CASE_TAGS, sample_tags=SAMPLE_TAGS
-    )
-    context.obj["api"] = AnalysisAPI(
+
+    context.obj["dna_api"] = AnalysisAPI(
         db=context.obj["db"],
-        hk_api=hk_api,
+        hk_api=hk.HousekeeperAPI(context.obj),
         tb_api=context.obj["tb"],
-        scout_api=scout_api,
-        lims_api=lims_api,
-        deliver_api=deliver,
-    )
-    context.obj["dna_api"] = MipAPI(
-        context.obj["mip-rd-dna"]["script"],
-        context.obj["mip-rd-dna"]["pipeline"],
-        context.obj["mip-rd-dna"]["conda_env"],
+        scout_api=scoutapi.ScoutAPI(context.obj),
+        lims_api=lims.LimsAPI(context.obj),
+        deliver_api=DeliverAPI(
+            context.obj,
+            hk_api=hk.HousekeeperAPI(context.obj),
+            lims_api=lims.LimsAPI(context.obj),
+            case_tags=CASE_TAGS,
+            sample_tags=SAMPLE_TAGS,
+        ),
+        script=context.obj["mip-rd-dna"]["script"],
+        pipeline=context.obj["mip-rd-dna"]["pipeline"],
+        conda_env=context.obj["mip-rd-dna"]["conda_env"],
+        root=context.obj["mip-rd-dna"]["root"],
     )
+    dna_api = context.obj["dna_api"]
 
     if context.invoked_subcommand is None:
```

Comment:
```suggestion
    if context.invoked_subcommand is not None:
        return
```
Then move everything below one indentation to left